extra/modules/greenbids-real-time-data/src/test/java/org/prebid/server/hooks/modules/greenbids/real/time/data/v1/GreenbidsRealTimeDataProcessedAuctionRequestHookTest.java

```diff
@@ -0,0 +1,589 @@
+import org.prebid.server.hooks.execution.v1.auction.AuctionRequestPayloadImpl;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.core.ThrottlingThresholds;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.model.data.GreenbidsInferenceDataService;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.model.predictor.FilterService;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.model.predictor.ModelCache;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.model.predictor.OnnxModelRunner;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.model.predictor.OnnxModelRunnerWithThresholds;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.model.predictor.ThresholdCache;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.model.result.AnalyticsResult;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.model.result.ExplorationResult;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.model.result.GreenbidsInvocationService;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.model.result.Ortb2ImpExtResult;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.v1.model.analytics.ActivityImpl;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.v1.model.analytics.AppliedToImpl;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.v1.model.analytics.ResultImpl;
+import org.prebid.server.hooks.modules.greenbids.real.time.data.v1.model.analytics.TagsImpl;
+import org.prebid.server.hooks.v1.InvocationAction;
+import org.prebid.server.hooks.v1.InvocationResult;
+import org.prebid.server.hooks.v1.InvocationStatus;
+import org.prebid.server.hooks.v1.analytics.Result;
+import org.prebid.server.hooks.v1.analytics.Tags;
+import org.prebid.server.hooks.v1.auction.AuctionInvocationContext;
+import org.prebid.server.hooks.v1.auction.AuctionRequestPayload;
+import org.prebid.server.json.JacksonMapper;
+import org.prebid.server.json.ObjectMapperProvider;
+import org.prebid.server.model.HttpRequestContext;
+import org.prebid.server.proto.openrtb.ext.request.ExtRequest;
+import org.prebid.server.proto.openrtb.ext.request.ExtRequestPrebid;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+import java.util.function.UnaryOperator;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class GreenbidsRealTimeDataProcessedAuctionRequestHookTest {
+
+    private GreenbidsRealTimeDataProcessedAuctionRequestHook target;
+
+    private JacksonMapper jacksonMapper;
+
+    private Cache<String, OnnxModelRunner> modelCacheWithExpiration;
+
+    private Cache<String, ThrottlingThresholds> thresholdsCacheWithExpiration;
```

Comment:
we usually mock dependencies

P.S. Waiting for the rest of unit tests

Reply:
Added new tests:
`FilterServiceTest`
- `shouldReturnFilteredBiddersWhenValidThrottlingMessagesProvided`
- `shouldThrowPreBidExceptionWhenOrtExceptionOccurs`
- `shouldThrowPreBidExceptionWhenThrottlingMessagesIsEmpty`
- `shouldThrowPreBidExceptionWhenTensorSizeMismatchOccurs`

`GreenbidsInferenceDataServiceTest`
- `shouldReturnValidThrottlingMessages`
- `shouldHandleMissingIp`
- `shouldThrowPreBidExceptionOnGeoIpFailure`

`GreenbidsInvocationServiceTest`
- `shouldReturnUpdateBidRequestWhenNotExploration`
- `shouldReturnNoActionWhenExploration`

`GreenbidsRealTimeDataModuleTest`
- `shouldHaveValidInitialConfigs`

`GreenbidsUserAgentTest`
- `getDeviceShouldReturnPCForWindowsNT`
- `getDeviceShouldReturnDeviceIPhoneWhenIOS`
- `getBrowserShouldReturnBrowserNameAndVersionWhenUserAgentIsPresent`
- `getBrowserShouldReturnEmptyStringWhenBrowserIsNull`

Reply:
As discussed on Slack, have applied `@ExtendWith(MockitoExtension.class)` annotation to test classes to mock dependencies